yt_dlp/extractor/vidio.py

```diff
@@ -150,3 +158,134 @@ def _real_extract(self, url):
+    _TESTS = [{
+        'url': 'https://www.vidio.com/live/204-sctv',
+        'info_dict': {
+            'id': '204',
+            'title': 'SCTV',
+            'uploader': 'SCTV',
+            'uploader_id': 'sctv',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        },
+    }, {
+        # Premier-exclusive livestream
+        'url': 'https://www.vidio.com/live/6362-tvn',
+        'only_matching': True,
+    }, {
+        # DRM premier-exclusive livestream
+        'url': 'https://www.vidio.com/live/6299-bein-1',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id, display_id = re.match(self._VALID_URL, url).groups()
+        stream_data = self._call_api(
+            'https://www.vidio.com/api/livestreamings/%s/detail' % video_id, display_id)
+        stream_meta = stream_data['livestreamings'][0]
+        user = stream_data.get('users', [{}])[0]
+
+        title = stream_meta.get('title')
+        username = user.get('username')
+
+        formats = []
+        if stream_meta.get('is_drm'):
+            # skip non-drm formats if drm is detected
+            if not self.get_param('allow_unplayable_formats'):
+                self.raise_no_formats(
+                    'This video is DRM protected.', expected=True)
+
+            drm_hls_url = stream_meta.get('drm_stream_hls_url')
+            if drm_hls_url:
+                formats.extend(self._extract_m3u8_formats(
+                    drm_hls_url, display_id, 'mp4', 'm3u8-native'))
+        elif stream_meta.get('is_premium'):
+            sources = self._download_json(
+                'https://www.vidio.com/interactions_stream.json?video_id=%s&type=livestreamings' % video_id,
+                display_id, note='Downloading premier API JSON')
+            if not (sources.get('source') or sources.get('source_dash')):
+                self.raise_login_required('This video is only available for registered users with the appropriate subscription')
+
+            if sources.get('source'):
+                token_json = self._download_json(
+                    'https://www.vidio.com/live/%s/tokens' % video_id,
+                    display_id, note='Downloading HLS token JSON', data=b'')
+                formats.extend(self._extract_m3u8_formats(
+                    sources['source'] + '?' + token_json.get('token', ''), display_id, 'mp4', 'm3u8_native'))
+        elif stream_meta.get('stream_token_url') or stream_meta.get('stream_dash_url'):
+            # prioritize token urls
+            token_hls_url = stream_meta.get('stream_token_url')
+
+            if token_hls_url:
+                token_json = self._download_json(
+                    'https://www.vidio.com/live/%s/tokens' % video_id,
+                    display_id, note='Downloading HLS token JSON', data=b'')
+                formats.extend(self._extract_m3u8_formats(
+                    token_hls_url + '?' + token_json.get('token', ''), display_id, 'mp4', 'm3u8_native'))
```

Comment:
```suggestion
            if sources.get('source'):
                token_json = self._download_json(
                    'https://www.vidio.com/live/%s/tokens' % video_id,
                    display_id, note='Downloading HLS token JSON', data=b'')
                formats.extend(self._extract_m3u8_formats(
                    sources['source'] + '?' + token_json.get('token', ''), display_id, 'mp4', 'm3u8_native'))
            elif sources.get('source_dash'):
                pass
            else:
                self.raise_login_required('This video is only available for registered users with the appropriate subscription')
        elif stream_meta.get('stream_token_url'):
            token_json = self._download_json(
                'https://www.vidio.com/live/%s/tokens' % video_id,
                display_id, note='Downloading HLS token JSON', data=b'')
            formats.extend(self._extract_m3u8_formats(
                stream_meta['stream_token_url'] + '?' + token_json.get('token', ''), display_id, 'mp4', 'm3u8_native'))
        elif stream_meta.get('stream_dash_url'):
            pass
```

Not a necessary change. Just that this code is cleaner imo. I am using `if dash: pass else:` instead of `if not dash:` so that it is easy to just replace `pass` with the relevant code in the future

Reply:
semi-done, though i used `if hls: if dash:` instead of `if hls: elif dash:` bc i wasn't sure if both should appear at the same time